### ISSUE TITLE
Author lint bug

### ISSUE THREAD

**Issue by felix91gr** [rust-lang/rust-clippy#3849]
I was playing with it today, and the compiler in my machine gave me an unexpected error. When I spotted it, it turned out to be a missing semicolon. Check out [this playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=86c48d89f8aaf407555f1a1cef0820e3), and run Clippy over it. You'll see the following `author` code:

```rust
if_chain! {
    if let StmtKind::Local(ref local) = stmt.node;
    if let Some(ref init) = local.init
    if let ExprKind::Call(ref func, ref args) = init.node;
    if let ExprKind::Path(ref path) = func.node;
    if match_qpath(path, &["std", "mem", "transmute"]);
    if args.len() == 1;
    if let ExprKind::Path(ref path1) = args[0].node;
    if match_qpath(path1, &["ZPTR"]);
    if let PatKind::Wild = local.pat.node;
    then {
        // report your lint here
    }
}
```

The second line is `if let Some(ref init) = local.init`. It's missing a semicolon!

So there's that. Thanks for the lint, btw. It's really helping me out ^^

**Comment by mikerite** [comment]
Wasn't this fixed by #3853?
